jvm/src/Language/Java.hs

```diff
@@ -469,19 +469,23 @@ class (SingI (Interp a), IsReferenceType (Interp a)) => Interpretation (a :: k)
 -- say, unmarshall a Java object to a Haskell value. Unlike coercing, in general
 -- reifying induces allocations and copies.
 class Interpretation a => Reify a where
+  -- | Invariant: the argument and the result are distinct. That is, the result
+  -- and the argument share no direct JVM object references.
```

Comment:
This is the only relevant condition:
```
The result and the argument share no direct JVM object references.
```
The values being distinct does not imply it.

Reply:
I'm not sure I get your point. I intended the second sentence as the definition of the word "distinct" in the first sentence. Said definition doesn't mention values. We could choose to remove introducing a definition entirely.

Reply:
I see. I find the meaning of distinct surprising here. It would be best to avoid it as you suggest.